src/SignalR/clients/csharp/Client/test/FunctionalTests/HubConnectionTests.cs

```diff
@@ -1335,6 +1335,96 @@ public async Task ServerThrowsHubExceptionIfBuildingAsyncEnumeratorIsNotPossible
             }
         }
 
+        [Theory]
+        [QuarantinedTest]
+        [MemberData(nameof(HubProtocolsList))]
+        public async Task ServerLogsErrorIfClientInvokeCannotBeSerialized(string protocolName)
+        {
+            // Just to help sanity check that the right exception is thrown
+            var exceptionSubstring = protocolName switch
+            {
+                "json" => "A possible object cycle was detected.",
+                "newtonsoft-json" => "A possible object cycle was detected.",
+                "messagepack" => "Failed to serialize Microsoft.AspNetCore.SignalR.Client.FunctionalTests.TestHub+Unserializable value.",
+                var x => throw new Exception($"The test does not have an exception string for the protocol '{x}'!"),
+            };
+
+            var protocol = HubProtocols[protocolName];
+            using (var server = await StartServer<Startup>(write => write.EventId.Name == "FailedWritingMessage"))
+            {
+                var connection = CreateHubConnection(server.Url, "/default", HttpTransportType.LongPolling, protocol, LoggerFactory);
+                var closedTcs = new TaskCompletionSource<Exception>();
+                connection.Closed += (ex) => { closedTcs.TrySetResult(ex); return Task.CompletedTask; };
+                try
+                {
+                    await connection.StartAsync().OrTimeout();
+
+                    var result = connection.InvokeAsync<string>(nameof(TestHub.CallWithUnserializableObject)).OrTimeout();
+
+                    // The connection should close.
+                    await closedTcs.Task.OrTimeout();
```

Comment:
Do we expect the Exception passed to the connection.Closed callback to be null? Or should it be some sort of serialization  exception? Should we verify that?